src/pages/[platform]/build-a-backend/data/custom-business-logic/batch-ddb-operations/index.mdx

```diff
@@ -0,0 +1,149 @@
+  platforms: [
+    'android',
+    'angular',
+    'flutter',
+    'javascript',
+    'nextjs',
+    'react',
+    'react-native',
+    'swift',
+    'vue'
+  ]
+};
+
+export const getStaticPaths = async () => {
+  return getCustomStaticPath(meta.platforms);
+};
+
+export function getStaticProps() {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+Batch DynamoDB operations allow you to add multiple items in single mutation.
+
+## Step 1 - Define a custom mutation
+
+```ts
+import { type ClientSchema, a, defineData } from '@aws-amplify/backend';
+
+const schema = a.schema({
+  // 1. Define your return type as a custom type or model
+  Post: a.model({
+    id: a.id(),
+    content: a.string(),
+    likes: a.integer()
+  }),
+
+  // 2. Define your mutation with the return type and, optionally, arguments
+  BatchCreatePost: a
+    .mutation()
+    // arguments that this query accepts
+    .arguments({
+      content: a.string().array()
+    })
+    .returns(a.ref("Post").array())
+    // only allow signed-in users to call this API
+    .authorization(allow => [allow.authenticated()])
+    .handler(
+      a.handler.custom({
+        dataSource: a.ref("Post"),
+        entry: "./BatchCreatePostHandler.js",
+      })
+    )
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema
+});
+```
+
+## Step 2 - Configure custom business logic handler code
+
+After your query or mutation is defined, you need to author your custom business logic using a [custom resolver powered by AppSync JavaScript resolver](https://docs.aws.amazon.com/appsync/latest/devguide/tutorials-js.html).
+
+Custom resolvers work on a "request/response" basis. You choose a data source, map your request to the data source's input parameters, and then map the data source's response back to the query/mutation's return type. Custom resolvers provide the benefit of no cold starts, less infrastructure to manage, and no additional charge for Lambda function invocations. Review [Choosing between custom resolver and function](https://docs.aws.amazon.com/appsync/latest/devguide/resolver-reference-overview-js.html#choosing-data-source).
+
+In your `amplify/data/resource.ts` file, define a custom handler using `a.handler.custom`.
+
+```ts title="amplify/data/resource.ts"
+import { type ClientSchema, a, defineData } from '@aws-amplify/backend';
+
+const schema = a.schema({
+  Post: a.model({
+    id: a.id(),
+    content: a.string(),
+    likes: a.integer()
+  }),
+
+  BatchCreatePost: a
+    .mutation()
+    .arguments({
+      contents: a.string().array()
+    })
+    .returns(a.ref("Post").array())
+    .authorization(allow => [allow.authenticated()])
+    // 1. Add the custom handler
+    .handler(
+      a.handler.custom({
+        dataSource: a.ref("Post"),
+        entry: "./BatchCreatePostHandler.js",
+      })
+    )
+});
+
+export type Schema = ClientSchema<typeof schema>;
+
+export const data = defineData({
+  schema
+});
+```
```

Comment:
Do we need to repeat the code block here? Seems redundant

Reply:
Oops, I meant to remove the `.handler` from the first block. I modeled this page on the existing custom resolver page. https://docs.amplify.aws/react/build-a-backend/data/custom-business-logic/